internal/k8s/reconciler/gateway.go

```diff
@@ -275,6 +328,9 @@ func (g *K8sGateway) isEqual(other *K8sGateway) bool {
 	if g.podReady != other.podReady {
 		return false
 	}
+	if g.serviceReady != other.serviceReady {
+		return false
+	}
```

Comment:
In addition to waiting for pods to be ready - which we were already doing - we now wait for the `Service` to be available and have an accessible IP before we mark the `Gateway` ready.

Reply:
Yeah, I think that's a good call since, esp in the case of something like a LoadBalancer-fronted implementation things aren't really ready until the service can be used.